### PR TITLE
Fix #1282 & #1433 - Organise folders and files and document their usage.

### LINKED ISSUE #1282: Organise folders and files and document their usage.
Currently the folder system in this website is a little muddled and could do with being tidied.
### Not externally accessible.

Any file or folder prefixed with a `.` or a `_` is not externally accessible.
- [x] `.github` is for special GitHub files only.
- [x] `.tx` is for special Transifex files only.
- [x] `_backend` is for storing backend files and libraries.
- [x] `_data` is for private data that is handled by some part of the site before being exposed.
- [x] `_lang` (previously `lang`) contains the translated Transifex files.
- [x] `_scripts` is for raw javascript, before being processed.
- [x] `_styles` is for raw stylesheets, before being processed.
- [x] `_templates` is for reusable sections of HTML.
- [x] `_tests` is for tests.
### Externally accessible.
- [x] `api` (previously `hook`) should contain hooks hit by other services, like printful, or ourselves.
- [x] `data` is for public data, like the slingshot json. This is loaded over HTTP.
- [x] `docs` is a site run by MDR, and should otherwise only contain `*.md` files.
- [x] `fonts` contains fonts we rely on.
- [x] `images` is for images.

### PR DESCRIPTION
**Pull request by lewisgoddard**
Fixes #1282 & #1433

### Changes Summary

- Migrate `lang` to `_lang`
- Migrate backend-only `data` to `_data`
- Move `l10n.php` from `_templates` to `_backend`
- Move `_templates/sitewide.php` to `_backend/preload.php`

### Warnings
At merge, we must:
- Clear caches
- `mv data/_average-payments.db _data/average-payments.db` and make sure the file _and_ the directory are writable.
- `touch data/average-payments.json`
- `touch data/chart.json`
- Set up cron for the files in `_backend/data-generators/*`
- Push and pull new translations.

---

This pull request is ready for review.

### CODE REVIEW

File: _backend/log-echo.php [hunk 1]
```diff
@@ -3,13 +3,13 @@
 // Setup sentry error logging
 if (isset($config['sentry_key']) && $config['sentry_key'] !== false) {
     $sentry = new Raven_Client($config['sentry_key']);
-    $sentry->install();
+    $toJSON->install();
```
Comment (btkostner) [inline _backend/log-echo.php]: Am I missing something here or should this still be `$sentry`?

**Comment by btkostner** [comment]
Just looked through all the comments, and it looks good (minus the one comment). I'm going to clone and try to check every page for regression before I approve though

**Comment by btkostner** [comment]
I approve this branch as it doesn't seem to break anything, but this will need to await infrastructure changes before pushing to master.

**Comment by btkostner** [comment]
I _really_ hope this doesn't break anything

**Comment by lewisgoddard** [comment]
I can't load https://elementary.io/data/average-payments.json and https://elementary.io/data/chart.json appears to be empty. Did you do all the "after merging" steps?

**Comment by btkostner** [comment]
The data generation steps are not yet put on a timer

**Comment by lewisgoddard** [comment]
@btkostner Any chance of getting that timer set up?

**Comment by btkostner** [comment]
@lewisgoddard Sorry about the delay, just updated the server with the needed services. Should be up and running here: https://elementary.io/data/average-payments.json

**Pull request merged**